packages/server-wallet/src/wallet/index.ts

```diff
@@ -563,7 +564,8 @@ export class Wallet extends EventEmitter<WalletEvent>
               return;
             case 'Withdraw':
               await this.store.addChainServiceRequest(action.channelId, 'withdraw', tx);
-              await this.chainService.concludeAndWithdraw(app.supported ? [app.supported] : []);
+              // Note, this returns a promise. We are NOT waiting on the promise
+              this.chainService.concludeAndWithdraw(app.supported ? [app.supported] : []);
```

Comment:
Is the ternary operator here just to appease the compiler? IE. is `app.supported` known to be defined at this point?
```suggestion
              this.chainService.concludeAndWithdraw([app.supported!]);
```

Reply:
If the wallet is functioning correctly, `app.supported` should be defined. Clarified here: 0bbd70baf.

Another option is to throw an error if app.supported is undefined. The non-null assertion you suggest also looks good, but, in general, I try to avoid those.

Reply:
But in this case, if `app.supported` happens to be `undefined`, the code is misbehaving, silently, which seems really bad.

At least the assertion would result in a runtime error. I agree with the rule that we should avoid them (or maybe not even have them).

Reply:
That reasoning makes sense to me.